### ISSUE TITLE
Archive failure No such module 'PPRiskMagnes'

### ISSUE THREAD

**Issue by ejensen** [braintree/braintree_ios#679]
**Integration Details (please complete the following information):**
 - Braintree SDK Version: 5.3.1
 - Environment: N/A
 - iOS Version: iOS 14.5 SDK
 - Device: arm64 devices
 - Integration type & version: Swift Package Manager 5.4.0

**Describe the bug**
Archiving fails when importing a subset of libraries within the Braintree Swift Package. The `PayPalDataCollector` library fails to compile: `No such module 'PPRiskMagnes'` 
This was not an issue in Braintree 5.3.0 and earlier.

**To Reproduce**
Steps to reproduce the behavior (video attached below):
1. Open the SPMTest app project
3. Remove a reference of the optional Braintree libraries (ex. Union Pay, Venmo, etc)
4. Set the archive build configuration to "Debug"
5. Perform a clean archive

**Expected behavior**
The archive completes successfully

**Screenshots**
<img width="526" alt="Screen Shot 2021-05-18 at 9 06 04 AM" src="https://user-images.githubusercontent.com/915431/118693260-c2d04e00-b7bf-11eb-92d8-3649d23f9c7b.png">

<img width="947" alt="Screen Shot 2021-05-18 at 9 52 33 AM" src="https://user-images.githubusercontent.com/915431/118693273-c6fc6b80-b7bf-11eb-8fff-3b6a7d668208.png">

https://user-images.githubusercontent.com/915431/118695280-de3c5880-b7c1-11eb-89cc-67dae7d06862.mp4

**Comment by JonathanDowning** [comment]
Seconded, I'm unable to update our project to 5.3.1 due to "No such module 'PPRiskMagnes'", it's not immediately obvious from [this diff](https://github.com/braintree/braintree_ios/compare/5.3.0...5.3.1) why this is now the case.

**Comment by scannillo** [comment]
👋  Hello. Thank you for reporting this issue. I will be taking a look at this today.

**Comment by scannillo** [comment]
With a fresh clone of the repo & Xcode 12.5 (Swift 5.4), I am not immediately able to replicate this issue. However, after doing a `clean`, I can replicate.

Listed in the [Swift 5.4 release notes](https://swift.org/blog/swift-5-4-released/) are some package caching updates. Here are some suggestions to get you up and running which worked for me:
- Remove and re-install all Braintree packages by deleting and re-adding the `braintree_ios` package in the `Swift Packages` section of your project settings.
- After cleaning, run / build the application itself and then attempt the archive again.

Let me know if those work!

**Comment by mccarron** [comment]
I'm also having the same exact problem with Xcode 12.5, can't do Archive builds at all. Tried the above steps and even rebooted a few times for good measure. 

Builds are fine, but Archives fail. Very strange really.

**Comment by scannillo** [comment]
Hi @mccarron - thank you for sharing. Would you mind giving some more info:
- Can you post the error logs you are seeing (to help us confirm what is going on)?
- Is this only a problem for you on Xcode 12.5 and Braintree iOS 5.3.1? 
- Can you share the command you are using to archive?
- Any other steps to replicate!

With a fresh clone of the `braintree_ios` repo, I am able to successfully archive the demo app after completing @ejensen's replication steps (removing `BraintreeUnionPay`) using both archive via Xcode (Product > Archive) and via command line (via `xcodebuild archive`).

**Comment by mccarron** [comment]
> Hi @mccarron - thank you for sharing. Would you mind giving some more info:
> 
> * Can you post the error logs you are seeing (to help us confirm what is going on)?

I can post this but I can't share full debug logs here. 

<img width="523" alt="Screen Shot 2021-05-24 at 10 38 20 AM" src="https://user-images.githubusercontent.com/45654/119372501-0114b400-bc7d-11eb-91f7-21be7f50f143.png">

> * Is this only a problem for you on Xcode 12.5 and Braintree iOS 5.3.1?

Correct this is Xcode 12.5 and Braintree iOS 5.3.1. It seems to archive just fine with Braintree iOS 5.3.0 on Xcode 12.5. 

I don't have Xcode 12.4 installed right now so I can't easily test if it's an Xcode issue as well as we're doing this SPM migration due to Xcode 12.5 fixing SPM issues. I may on another machine so I can report back later today.

> * Can you share the command you are using to archive?

Product > Archive as you said and we also use `fastlane` to Archive also fails on our CI environment. It generates a long command that is not relevant here and can't share all of our debug logs as I said above. 

> * Any other steps to replicate!

I'll try to reproduce on a clean project, it seems fairly reliable to do here with little work needed to reproduce for us. 

We also use Braintree Drop In so think it may be related to the combo use maybe?

For now we'll stay on 5.3.0 since that seems to work just fine.

**Comment by mccarron** [comment]
Quick update it does in fact fail only on Xcode 12.5 but works fine on Xcode 12.4, switched between them and triggered the failure only on Braintree iOS 5.3.1.

> CompileSwift normal arm64 /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/SourcePackages/checkouts/braintree_ios/Sources/PayPalDataCollector/Public/PayPalDataCollector/PPDataCollector.swift (in target 'PayPalDataCollector' from project 'Braintree')
>     cd /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/SourcePackages/checkouts/braintree_ios
>     /Applications/Xcode\ 12.5.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -emit-bc /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/SourcePackages/checkouts/braintree_ios/Sources/PayPalDataCollector/Public/PayPalDataCollector/PPDataCollector.swift -emit-module-path /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/Objects-normal/arm64/PayPalDataCollector.swiftmodule -emit-module-doc-path /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/Objects-normal/arm64/PayPalDataCollector.swiftdoc -emit-module-source-info-path /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/Objects-normal/arm64/PayPalDataCollector.swiftsourceinfo -serialize-diagnostics-path /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/Objects-normal/arm64/PayPalDataCollector-master.dia -emit-objc-header-path /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/Objects-normal/arm64/PayPalDataCollector-Swift.h -emit-dependencies-path /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/Objects-normal/arm64/PayPalDataCollector-master.d -target arm64-apple-ios12.0 -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode\ 12.5.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk -I /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/BuildProductsPath/Local-iphoneos -I /Applications/Xcode\ 12.5.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/lib -F /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/BuildProductsPath/Local-iphoneos -F /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/BuildProductsPath/Local-iphoneos/PackageFrameworks -F /Applications/Xcode\ 12.5.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -F /Applications/Xcode\ 12.5.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk/Developer/Library/Frameworks -g -module-cache-path /Users/mccarron/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity\=checked -O -D SWIFT_PACKAGE -D Xcode -serialize-debugging-options -Xcc -working-directory -Xcc /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/SourcePackages/checkouts/braintree_ios -Xcc -I/Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/swift-overrides.hmap -Xcc -I/Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/BuildProductsPath/Local-iphoneos/include -Xcc -I/Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/DerivedSources-normal/arm64 -Xcc -I/Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/DerivedSources/arm64 -Xcc -I/Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/DerivedSources -Xcc -DSWIFT_PACKAGE -target-sdk-version 14.5 -parse-as-library -module-name PayPalDataCollector -num-threads 8 -o /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/Build/Intermediates.noindex/ArchiveIntermediates/clientLocal/IntermediateBuildFilesPath/Braintree.build/Local-iphoneos/PayPalDataCollector.build/Objects-normal/arm64/PPDataCollector.bc
> 
> /Users/mccarron/Library/Developer/Xcode/DerivedData/client-gzkndxhzcubkocfsukhbcquncyig/SourcePackages/checkouts/braintree_ios/Sources/PayPalDataCollector/Public/PayPalDataCollector/PPDataCollector.swift:1:8: error: no such module 'PPRiskMagnes'
> import PPRiskMagnes
>

**Comment by scannillo** [comment]
Thank you very much for providing that detail! 

I created a new sample app via a blank Xcode project, included Braintree Drop In v9.0.1 (with Braintree 5.3.1) via SPM. I was able to replicate your archive issue via the XCode UI "Product > Archive". 

However, I then ran `xcodebuild -project 'TestApp.xcodeproj' -scheme 'TestApp' clean archive CODE_SIGNING_ALLOWED=NO` and was able to successfully archive on the command line. Then, running archive in the Xcode UI also succeeded. Definitely weird, and we're still trying to figure out more details / what is going on. But wanted to update you! Feel free to give it a try.

**Comment by rursache** [comment]
i'm having the same issue on xcode 11.5 - No such module 'PPRiskMagnes' - even on running the app. 

i only need the braintree + paypal integration and it seems that PPRiskMagnes lib is still creating issues (last time i tried this integration it was in february).

sometimes after multiple builds the app runs but i get a runtime crash:
```swift
dyld: Library not loaded: @rpath/PPRiskMagnes.framework/PPRiskMagnes
  Referenced from: /private/var/containers/Bundle/Application/0C2D8DA4-D8DB-4086-8E32-65B42C27799B/<APPNAME>.app/<APPNAME>
  Reason: image not found
dyld: launch, loading dependent libraries
DYLD_LIBRARY_PATH=/usr/lib/system/introspection
DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
```

what i tried:
- cleaning the project about 1233923 times
- removing and re-adding the package around the same amount of times
- version 5.3.0, 5.3.1, 5.3.2
- multiple branches (master, test-xcframework-zips, spm-binary-fix-xcode12-5)
- creating an empty project with the same braintree targets
- not checking PPRiskMagnes when adding braintree and manually linking + embed & sign of that PPRiskMagnes.xcframework

Braintree is the only dependency that still keeps me from fully migrating from cocoapods to SPM. This is really annoying because all the other "big guys" already got SPM figured out (Stripe, Firebase, etc).

**Comment by JetForMe** [comment]
@sestevens mentioned in #576 that "cleaning and re-building" solves the issue. The thing is, at least for me, I never have an issue with Build, but only with Archive. AFAICT, Archive is always a clean-slate build.

**Comment by simonpierreroy** [comment]
This seems to be a bug in the Xcode build system for swiftPM integration! Try this https://github.com/simonpierreroy/xcframework-issue and always clean in between it should fail! It does the same with my own local package that has an XCframework! Here is my feedback `FB9125479 `. I used this repo for my feedback to apple, since I was having the same issues with close source package I can not share.

<img width="1098" alt="Screen Shot 2021-06-02 at 12 18 26 PM" src="https://user-images.githubusercontent.com/44030728/120515806-afff7100-c39c-11eb-902b-1c538c634ae3.png">

**Comment by JackoPlane** [comment]
While we wait for an update to SPM, Is there an older version of the Braintree SDK that can be used with SPM that doesn't rely on `PPRiskMagnes` ?

**Comment by scannillo** [comment]
Hi everyone. We think this is likely [an Xcode bug](https://bugs.swift.org/browse/SR-13803), but are still doing some testing internally. Like I said in my previous posts, we were only able to trigger the archive issues _sometimes_. Building & running the application, re-installing the package via SPM, and archiving on the command line all got things working for us.

@JackoPlane - people in this thread are saying that they _don't_ see archive issues on 5.3.0, however @sestevens and I were seeing some similar issues in our CI near that version release. We can't say for sure that the issue is not present on 5.3.0, but you can give it a shot for now.

**Comment by mccarron** [comment]
> @JackoPlane - people in this thread are saying that they _don't_ see archive issues on 5.3.0, however @sestevens and I were seeing some similar issues in our CI near that version release. We can't say for sure that the issue is not present on 5.3.0, but you can give it a shot for now.

Agree with ^ I experienced it once with 5.3.0 locally but went away with a clean, but no issues on CI at all so far. *knocks on wood*

**Comment by JetForMe** [comment]
I do believe it's an Xcode bug, and only once did I succeed in archiving. Dozens of attempts later with all manner of cleaning, resetting, re-downloading, removing and re-adding, I haven't been able to Archive. I submitted our project to Apple, and it looks like another Apple engineer has taken on the Swift [bug](https://bugs.swift.org/browse/SR-13803) for it. I added @simonpierreroy's info to that.

We’re using 5.3.0.

**Comment by JetForMe** [comment]
One possibility might be if Braintree packaged up the entire suite as a single binary framework (since I presume there are parts of your code you don't want to share). We might be able to link against that. I think it's the transitive dependency that fails.

**Comment by RamblinWreck77** [comment]
Just chiming to confirm I am seeing this same error in:

Apple Silicon
Xcode 12.5
Braintree_ios 5.4.0

In debug OR release mode:
-Clean
-Build/Run attempt #1 (always fails)
-Build/Run attempt #2 (sometimes fails)
-Build/Run attempt #3 (always works)

I was able to work around this issue by:

Set Release mode/any iOS device
1: clean
2: Build (always fails)
3: Build (might work, if it works go to 5)
4: Build again (should always work)
5: Archive (should spit out a binary now)

**Comment by RamblinWreck77** [comment]
Also, this same problem totally breaks SwiftUI previews for the entire project which is super frustrating. Unfortunately I have not been able to find a workaround/retry pattern that gets SwiftUI previews to work. 

Back to the dark ages of code -> compile -> test on device -> tweak code again 😂

**Comment by eseay** [comment]
I ran into this issue with BT 5.3.0 and 5.4.0, but I have downgraded to 5.2.0, and that seems to be working for me.

**Comment by scannillo** [comment]
> Also, this same problem totally breaks SwiftUI previews for the entire project which is super frustrating. Unfortunately I have not been able to find a workaround/retry pattern that gets SwiftUI previews to work.

Hi @RamblinWreck77 - `braintree_ios` doesn't offer any UI (besides for the payment methods that perform an app/browser switch). If you're referring to `braintree-ios-drop-in`, feel free to open an issue regarding the SwiftUI problem in that repo.

**Comment by JetForMe** [comment]
> Also, this same problem totally breaks SwiftUI previews for the entire project which is super frustrating. Unfortunately I have not been able to find a workaround/retry pattern that gets SwiftUI previews to work.
> 
> Back to the dark ages of code -> compile -> test on device -> tweak code again 😂

Unrelated to braintree, almost certainly, but I've run into SwiftUI preview problems that say things like "<somefile>.swift is not a member of any target." I found removing and re-adding that file to the project resets Xcode's notions.

**Comment by scannillo** [comment]
👋  Hello everyone. We pushed a potential "fix" to [a branch called `xcode-archive-test`](https://github.com/braintree/braintree_ios/tree/xcode-archive-test). With this change, we're no longer experiencing any archive issues with PPRiskMagnes via SPM. Feel free to give this branch a try and let us know if it resolves your issue.

**Comment by rursache** [comment]
> 👋 Hello everyone. We pushed a potential "fix" to [a branch called `xcode-archive-test`](https://github.com/braintree/braintree_ios/tree/xcode-archive-test). With this change, we're no longer experiencing any archive issues with PPRiskMagnes via SPM. Feel free to give this branch a try and let us know if it resolves your issue.

i'm still getting `No such module 'PPRiskMagnes'`. am i missing something? thanks!

**Comment by scannillo** [comment]
Hi @rursache - remove the dependency, clean, re-build etc?

**Comment by rursache** [comment]
> Hi @rursache - remove the dependency, clean, re-build etc?

exactly in that order ☹️
builds fine but does not archive

**Comment by JetForMe** [comment]
I spent time in an Xcode lab during WWDC. They fully acknowledged the bug and will be working on it, although there's no indication as to when it will be fixed (hopefully they'll do an Xcode 12.5.1!).

In the meantime, they offered this workaround, which I have not had a chance to try yet. The failure is caused by a clean build failing to properly process XCFrameworks. This processing step involves copying them to the correct location in the built products directory. The proposed workaround is to manually copy the binary XCFrameworks to the appropriate location in the build products directory. This might work as a Run Script build phase in the main target, or you might need to create a separate Aggregate target and make that a dependency of the target you're trying to build.

The correct location can be found by doing one of the target builds that works and looking for where it copies `PPRiskMagnes`. Unfortunately I can't take the time right now to figure that out.

**Comment by simonpierreroy** [comment]
Same here :) my Feedback was acknowledge and I was told during a WWDC lab they are working on it. No timeline. (I hope a beta of Xcode 13).  They confirm my hypothesis for the cause described here (at least told me it was a build system issue) https://github.com/braintree/braintree_ios/issues/679#issuecomment-853166124.

**Comment by scannillo** [comment]
Thank you @JetForMe & @simonpierreroy for communicating this issue to Apple dev's at WWDC 👍 . 

Is anyone hard blocked by this issue? If so, please let us know.

On our end - we are always able to circumvent this bug by cleaning, then re-building/ running the actual application, and then archiving succeeds.

**Comment by JetForMe** [comment]
We're not blocked only because we're still using CocoaPods to integrate it, and we have a crucial deadline that's taking priority over the goal of getting rid of CocoaPods. In six weeks, we'll see.

**Comment by mthole** [comment]
I am regularly able to reproduce this issue on our CI machines, even with the `xcode-archive-test` branch. Note that I'm not even trying to archive, just do a normal Debug build, e.g.:

```
xcodebuild -workspace MyApp.xcworkspace -scheme MyApp -configuration Debug CODE_SIGNING_ALLOWED=NO COMPILER_INDEX_STORE_ENABLE=NO build | xcbeautify
```

**Comment by scannillo** [comment]
> Silly question: Should the Package.swift simply specify that the PayPalDataCollector target depends on PPRiskMagnes? That doesn't seem to be the case today.

Great question! That is how it should be, but due to an Xcode/SPM bug when a library depends on a binary target, we had to expose PPRiskMagnes as its own library as a workaround. You can see that discussion [here](https://github.com/braintree/braintree_ios/issues/576#issuecomment-759034149). The need for this workaround is apparently solved in Xcode 12.5, and we are currently working on / testing that update now.

I'm glad you have a workaround for CI. Today we will release a version of the SDK with a workaround that has calmed our CI from seeing this bug so frequently as well. Will keep you posted!

**Comment by scannillo** [comment]
👋  Hi everyone. We released [Braintree v5.4.1](https://github.com/braintree/braintree_ios/releases/tag/5.4.1) with a workaround that has consistently removed this issue from both our local testing and CI builds. We understand this issue is some combination of an Xcode bug with our SDK - but wanted to release this fix to help resolve this issue for you, if possible.

Thank you all for thorough feedback and help! Feel free to keep us posted by commenting, or opening a new issue if problems arise.

**Comment by mccarron** [comment]
An FYI we were still seeing this with `5.4.1` and `5.4.2` so we're rolling back to `5.3.0` which didn't have this issue for us.

Not sure if happening for others as well?

**Comment by mthole** [comment]
I am still hitting this issue after upgrading to `5.4.2` and removing the explicit dependencies on `PPRiskMagnes`, etc.

![image](https://user-images.githubusercontent.com/57376542/124793795-33176800-df03-11eb-9efb-06db5dbe3931.png)

cc: @scannillo @mccarron FYI

**Comment by rursache** [comment]
> I am still hitting this issue after upgrading to `5.4.2` and removing the explicit dependencies on `PPRiskMagnes`, etc.

you need `PPRiskMagnes` tho

**Comment by mthole** [comment]
> > @mthole: I am still hitting this issue after upgrading to `5.4.2` and removing the explicit dependencies on `PPRiskMagnes`, etc.
> 
> you need `PPRiskMagnes` tho

The release notes say:

```
You can remove the KountDataCollector, PPRiskMagnes, and CardinalMobile explicit dependencies.
```

I read that to mean I should remove the explicit deps I have on `PPRiskMagnes`, and that SPM now transitively links them now as needed? For context, I am adding `BraintreePayPal` as the dependency to my project.

**Comment by rursache** [comment]
the [installation guide](https://github.com/braintree/braintree_ios/blob/master/SWIFT_PACKAGE_MANAGER.md) is **very** clear around what you need to include when targeting parts of the sdk. your issue is not related to #679.

**Comment by mthole** [comment]
@rursache Doesn't look like that installation guide has been updated for `5.4.2`, which contains the changes I'm referencing.

**Comment by mthole** [comment]
Got my set up working! For anyone else: After updating to `5.4.2`, I removed my `PPRiskMagnes` direct dependency (correctly), but I also removed the `PayPalDataCollector` dependency, which is still needed.

**Comment by scannillo** [comment]
@rursache is correct - for v5.4.2+, you no longer need to explicitly include `PPRiskMagnes`, `KountDataCollector`, or `CardinalMobile`. As of now, you **do** still need to include `PayPalDataCollector` for PayPal, 3DS, and Local Payments/

**Comment by JetForMe** [comment]
We're able to build with Card, Core, ApplePay, and PayPal, and haven't needed PayPalDataCollector in order to build and archive. But we are seeing *very* bizarre behavior with the iOS drop-in UI not displaying the vault (stored methods), except for one build we did earlier today, and we can't figure out why.

**Comment by simonpierreroy** [comment]
@scannillo could you explain the change you did in the framework and your finding to make it work (why Xcode 12.5 for example) :) it would be useful for one of my project. Thank you so much.

**Comment by scannillo** [comment]
Hi @simonpierreroy - sure, happy to share. Check out [this PR](https://github.com/braintree/braintree_ios/pull/636), in which the description outlines the situation and solution more thoroughly.
